src/components/helpers/touch_scroll_hook.ts

```diff
@@ -0,0 +1,92 @@
+import { Ref } from "../../types";
+import { useRefListener } from "./listener_hook";
+
+const friction = 0.95;
+
+const verticalScrollFactor = 1;
+
+const horizontalScrollFactor = 1;
+
+export function useTouchScroll(
+  ref: Ref<HTMLElement>,
+
+  updateScroll: (offsetX: number, offsetY: number) => void,
+
+  canMoveUp: () => boolean
+) {
+  let lastX = 0;
+  let lastY = 0;
+  let velocityX = 0;
+  let velocityY = 0;
+  let isMouseDown = false;
+  let lastTime = 0;
+
+  useRefListener(ref, "touchstart", onTouchStart, { capture: false });
+  useRefListener(ref, "touchmove", onTouchMove, { capture: false });
+  useRefListener(ref, "touchend", onTouchEnd, { capture: false });
+
+  function clientValue(ev: MouseEvent | TouchEvent) {
+    const clientX = ev instanceof MouseEvent ? ev.clientX : ev.touches[0].clientX;
+    const clientY = ev instanceof MouseEvent ? ev.clientY : ev.touches[0].clientY;
```

Comment:
in which case is this a MouseEvent or a TouchEvent ?

Reply:
good question, it should be a touchevent 100% of the time so that helper doesn't really make sense .. i'll loot into it

Reply:
@LucasLefevre I removed the helper, it really was useless